docs/Makefile

```diff
@@ -1,41 +1,91 @@
 DOCBOX=docs-buildbox:latest
+# BUILDROOT needs to be relative so that it works inside & outside the docbox container
+# - build to the gravity version specific version directory if running under the uber-make
+# - build to ../build/docs if running in this directory
+BUILDROOT ?= $(if $(GRAVITY_VERSION),../build/$(GRAVITY_VERSION),../build)
+BUILDDIR:=$(BUILDROOT)/docs
+DOCBOX_ID_FILE=$(BUILDROOT)/.docs-buildbox.id
 HOME=/home
 PORT=6601
 DOCKERPORTS = -p $(PORT):$(PORT)
-DOCKERFLAGS = --rm=true -v "$$(pwd)/../":$(HOME) -w $(HOME) -h docs
+DOCKERFLAGS = --rm=true -v "$$(pwd)/../":$(HOME) -w $(HOME)/docs -h docs $(DOCKERPORTS)
 NOROOT=-u $$(id -u):$$(id -g)
+DOCKERMAKE=docker run $(NOROOT) $(DOCKERFLAGS) $(DOCBOX) make BUILDROOT=$(BUILDROOT)
+VERSIONS=4.x 5.x 6.x 7.x
+LATEST=$(lastword $(VERSIONS))
+CFGS=$(addsuffix .yaml, $(VERSIONS))
+LATEST_CFG=$(lastword $(CFGS))
+RUN_CFG?=$(LATEST_CFG)
 
 # generate user-facing documentation
 .PHONY:docs
-docs: bbox
-	@docker run $(NOROOT) $(DOCKERFLAGS) $(DOCBOX) docs/build.sh
+docs: $(DOCBOX_ID_FILE)
+	$(DOCKERMAKE) container-compile-docs
 
+# run static analysis against the docs
+.PHONY: lint
+lint: $(DOCBOX_ID_FILE)
+	$(DOCKERMAKE) container-lint
 
-# 'make run' starts running the docs
+# run the docs in development mode
 .PHONY:run
-run: bbox
-	@echo "\n\nYou have to open http://localhost:$(PORT)/overview/ in your local browser\n\n"
-	docker run  $(NOROOT) $(DOCKERFLAGS) $(DOCKERPORTS) $(DOCBOX) \
-		 docs/build.sh run
+run: $(DOCBOX_ID_FILE) $(RUN_CFG)
+	@echo "\n\nOpen http://localhost:$(PORT)/ in your local browser\n\n"
+	$(DOCKERMAKE) container-run RUN_CFG=$(RUN_CFG)
 
-# install mkdocs on this machine 
-.PHONY:mkdocs
-mkdocs:
-	pip install -q -U mkdocs
+# drops you into bash shell of mkdocs container
+.PHONY:shell
+shell: $(DOCBOX_ID_FILE)
+	docker run -ti $(NOROOT) $(DOCKERFLAGS) $(DOCKERPORTS) $(DOCBOX) \
+		/bin/bash
 
-# builds a docker container which is used for running `mkdocs`
-.PHONY:bbox
-bbox:
+.PHONY:docbox
+$(DOCBOX_ID_FILE): Dockerfile
+	mkdir -p $(BUILDDIR)
 	docker build \
 		--build-arg UID=$$(id -u) \
 		--build-arg GID=$$(id -g) \
 		--build-arg HOME=$(HOME) \
```

Comment:
Does this need the WORKDIR change from above?  It doesn't seem to matter to the end user.

Reply:
fixed in ee1376f